plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/HiveMetastore.java

```diff
@@ -62,10 +62,20 @@ default void updatePartitionStatistics(Table table, String partitionName, Functi
 
     List<String> getAllTables(String databaseName);
 
+    /**
+     * @return List of table names from all schemas or Optional.empty if operation is not supported
+     */
+    Optional<List<SchemaTableName>> getAllTables();
+
     List<String> getTablesWithParameter(String databaseName, String parameterKey, String parameterValue);
 
     List<String> getAllViews(String databaseName);
 
+    /**
+     * @return List of view names from all schemas or Optional.empty if operation is not supported
```

Comment:
Can this include materialized views? let's document

Reply:
Updated